crates/red_knot/src/symbols.rs

```diff
@@ -79,15 +80,47 @@ impl Scope {
     }
 }
 
+#[derive(Debug)]
+pub(crate) enum Disposition {
```

Comment:
naming nit, but I'd probably go for `Kind` here. Both `Kind` and `Disposition` seem equally generic (as in, `Disposition` doesn't communicate useful additional context, at least to me), and `Kind` is shorter :)

But if you have arguments for preferring `Disposition`, I'm all ears.